docs/data-structures.md

```diff
@@ -0,0 +1,251 @@
+    "header": {
+        ...
+    },
+    "proposal_transactions": []
+    }
+]
+}
+```
+
+### Description
+
+#### Block
+
+| Name                    | Type            | Description                                                  |
+| ----------------------- | --------------- | ------------------------------------------------------------ |
+| `header`                | `Header`        | **The block header of the block.** This part contains some metadata of the block. |
+| `commit_trasactions`    | [`Transaction`] | **An array of transactions contained in the block.** This is where the miner put the received transactions. |
+| `proposal_transactions` | [string]        | **An array of hex-encoded short transaction ID.**            |
+| `uncles`                | [`UncleBlock`]  | **An array of uncle blocks of the block.**                   |
+
+#### Header
+
+| Name           | Type                | Description                                                  |
+| -------------- | ------------------- | ------------------------------------------------------------ |
+| `cellbase_id`  | H256(hash)          | **The hash of the Cellbase transaction.** Cellbase transaction is just like the coinbase transaction in Bitcoin. It's the transaction added by the miner who mined this block, by which the miner receives block reward for successfully mined the block. |
+| `difficulty`   | Bytes               | **The difficulty of the PoW puzzle.**                        |
+| `hash`         | H256(hash)          | **The block hash.** Note that although this value is returned by the JSON-RPC API, it is in fact NOT a part of the header. This value is actually calculated from the `Header` data. |
+| `number`       | uint64              | **The block height.**                                        |
+| `parent_hash`  | H256(hash)          | **The hash of the parent block.**                            |
+| `seal`         | `nonce` and `proof` | **The seal of a block.** After finished the block assembling, the miner can start to do the calculation for finding the solution of the PoW puzzle. The "solution" here is called `seal`. |
+| `seal.nonce`   | uint64              | **The nonce.** Similar to [the nonce in Bitcoin](https://en.bitcoin.it/wiki/Nonce). |
+| `seal.proof`   | Bytes               | **The solution of the PoW puzzle.**                          |
+| `timestamp`    | uint64              | **A [Unix time](http://en.wikipedia.org/wiki/Unix_time) timestamp.** |
+| `txs_commit`   | H256(hash)          | **The Merkle Root of the Merkle trie with the hash of transactions as leaves.** |
+| `txs_proposal` | H256(hash)          | **The Merkle Root of the Merkle trie with the hash of short transaction IDs as leaves.** |
+| `uncles_count` | uint32              | **The number of uncle blocks.**                              |
+| `uncles_hash`  | H256(hash)          | **The hash of the serialized uncle blocks data.** This will later be changed to using Merle Trie as well. |
```

Comment:
The serialization will be changed. Will not use Merle Trie.

Reply:
thx!
done